backend/src/main/java/cz/cuni/mff/xrg/odcs/backend/i18n/Messages.java

```diff
@@ -43,6 +44,7 @@ public static String getString(final String key, final Object... args) {
     private static ReloadableResourceBundleMessageSource initializeResourceBundle() {
         ReloadableResourceBundleMessageSource ms = new ReloadableResourceBundleMessageSource();
         ms.setDefaultEncoding("UTF-8");
+        ms.setFallbackToSystemLocale(false);
```

Comment:
EN locale is not available, that is the mistake you live in.
We always have
```
messages_sk.properties
messages.properties
```
so there is SK locale available and one default property source.
Lets have machine with SK locale in OS. In frontend.properties we have 
"EN" as the setting:
In Java's ResourceBundle(EN), the default behavior is to search:
1. messages_en.properties
2. messages_{what has OS set = sk} -> so messages_sk.properties
3. messages.properties

Same behavior is copied by Spring with "fallbackToSystemLocale(true)" 
which is the default. Yes, this is something you generally do not want, 
but they copied the behavior from ResourceBUndle, to let the world be 
consistent on this.

So to enjoy this default behavior, you would have to shipp:
```
messages_en.properties
messages_sk.properties
messages.properties
```
with your application, where messages_en.properties are same as 
messages.properties. Something you definitely do not want to do.
So they have added an option to start behave rationally, as everyone 
does, and ship only
```
messages_sk.properties
messages.properties
```
where fallbackToSystemLocale(false) skips step 2. from the previously 
described search algorithm:
1. messages_en.properties
2. SKIPPED messages_{what has OS set = sk} -> so messages_sk.properties
3. messages.properties
and this works as you expect, when it finds a language, it uses that 
language, when not, it uses default messages.properties and does not try 
to search for system specific locale - ignoring the value we have in 
configuration file. We want our app to be deterministic - to follow this 
algorithm of search, not the Java one (bad one) which is 
non-deterministic with regards on deployment environment (never mind 
what is in configuration, i will use what is in system environment).

Java has more of these tricky places (e.g. String.getBytes()).

On 07/13/2015 09:06 AM, Tomas Knap wrote:
>
> In 
> backend/src/main/java/cz/cuni/mff/xrg/odcs/backend/i18n/Messages.java 
> <https://github.com/UnifiedViews/Core/pull/445#discussion_r34437331>:
>
> > @@ -43,6 +44,7 @@ public static String getString(final String key, final Object... args) {
> >      private static ReloadableResourceBundleMessageSource initializeResourceBundle() {
> >          ReloadableResourceBundleMessageSource ms = new ReloadableResourceBundleMessageSource();
> >          ms.setDefaultEncoding("UTF-8");
> > +        ms.setFallbackToSystemLocale(false);
>
> Question: If it falls back to system locale, this should be only after 
> it tries to get proper bundle based on settings of locale in 
> config.properties, right? So the question is how the situation in #444 
> <https://github.com/UnifiedViews/Core/issues/444> may happen? If 
> Michal sets up "en" as default locale, fallback locale should never be 
> used anyway (because default en locale is always available), am I 
> right? So is it a bug in Spring? Or how it is possible that sk is used 
> when en is set up in config.properties and available always?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/UnifiedViews/Core/pull/445/files#r34437331>.
>